### ISSUE TITLE
Problems with Mappable and pluck

### ISSUE THREAD

**Issue by sebastian-zarzycki-es** [jarektkaczyk/eloquence#105]
Consider following:
```
$uniqueFreightCarriersWorking = Shipment::distinct()->select('freight_carrier')->get();
$uniqueFreightCarriersWorking = $uniqueFreightCarriersWorking->pluck('freightCarrier')->toArray();
```

this works (giving me an array of column values for freight_carrier property).

```
$uniqueFreightCarriersNotWorking = Shipment::distinct()->select('freightCarrier')->get();
$uniqueFreightCarriersNotWorking = $uniqueFreightCarriersNotWorking->pluck('freightCarrier')->toArray();
```

This doesn't (note the select('freightCarrier')) - while it pulls the right model and its value, after I use pluck, I just get null. Mapping is set properly in model:

``
  protected $maps = [
     'freightCarrier' => 'freight_carrier'
];
``

The documentation says "Note that MorphTo mapping doesn't support join-based querying (orderBy, pluck and aggregates)." - is this related? I'm not exactly sure what it means. So are there instances in which I cannot use mapped attributes?

**Comment by jarektkaczyk** [comment]
The only problem here is that `select('mapping')` results in `select('mapped as mapping')`, so later after the query returns models, the attribute set on the model is `mapping` rather than `mapped` (so Mappable doesn't have the value to work with..).

Not sure which way to go with it, but for now I can suggest something between the lines:
```
// assuming L5.2, in 5.3 skip collect()
$carriers = collect(Shipment::distinct()->select('freight_carrier')->getQuery()->get())->pluck('freightCarrier')->all();
```

**Comment by jarektkaczyk** [comment]
It's fixed in `5.3.1`:

```php
>>> $mapped = new class extends \Illuminate\Database\Eloquent\Model {
...       use \Sofa\Eloquence\Eloquence, Sofa\Eloquence\Mappable; 
...       protected $table = 'users'; 
...       protected $maps = ['nice' => 'email']; 
...    }
=> Illuminate\Database\Eloquent\Model@anonymous {#715}

>>> $user = $mapped::select('id', 'nice')->first()
=> Illuminate\Database\Eloquent\Model@anonymous {#723
     id: "1",
     email: "catalina.mayer@example.org",
   }
>>> $user->nice
=> "catalina.mayer@example.org"

>>> $users = $mapped::select('id', 'nice')->get()
=> Illuminate\Database\Eloquent\Collection {#733 ... }

>>> $users->pluck('nice', 'id')
=> Illuminate\Support\Collection {#676
     all: [
       1 => "catalina.mayer@example.org",
       8 => "cyrus84@example.org",
       5 => "gretchen.buckridge@example.org",
       7 => "johnathan49@example.net",
       3 => "kip.ebert@example.org",
       9 => "koby.donnelly@example.com",
       4 => "leila.wuckert@example.org",
       2 => "lucio.ondricka@example.com",
       10 => "wmetz@example.net",
       6 => "xkohler@example.com",
     ],
   }
```
